File: Program/1_Array/Array Standard problem/easy/2_find_second_largest_element.c
#include<stdio.h>

void main()
{
    int arr[10] = {10,12,45,17,21,57,33};
    int size = sizeof(arr)/sizeof(int);
    int i;
    int max = 0;
    for (i=0; i<size; i++)
        {
            if(arr[i] > max)
                {
                    max = arr[i];
                }
        }
    int j;
    printf("%d\n", max);
    int second_max=0;
    for(j=0; j<size; j++)
        {
            if(arr[j] != max)
                {
                    if(arr[j] > second_max)
                        {
                            second_max = arr[j];
                        }
                }
        }
    printf("The second largest element is :%d", second_max);

}

File: Program/1_Array/Array Standard problem/easy/2_find_second_largest_element.py
def find_second_largest(arr):
    max_element = max(arr)
    second_max = arr[0]
    for i in range(len(arr)):
        if arr[i] > second_max and arr[i] != max_element:
            second_max = arr[i]
    
    return second_max


l = [10,15,80,65,189,256,36,563,256,782]
print(find_second_largest(l))
